### PR TITLE
feat(swap): add swap inputs components

### PR DESCRIPTION
**Pull request by nistadev**
**Screenshots**

image

### CODE REVIEW

**Comment by neuodev** [comment]
[autoscirpt] Jira Ticket: **[[swap] Create the "From" and "To" field component](https://emurgo.atlassian.net/browse/YOEXT-650)**

[](__JIRA_COMMENT_FLAG)

File: packages/yoroi-extension/app/components/swap/SwapInput.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            >
+              MAX
+            </Typography>
+          </Box>
+        ) : (
+          <Box />
+        )}
+        <Box sx={{ justifySelf: 'end', alignSelf: 'end' }}>
+          <Typography variant="caption" color="#6B7384">
+            Current balance: {amount} {ticker}
```
Comment (neuodev) [inline packages/yoroi-extension/app/components/swap/SwapInput.js:134]: Please use `Intl`

Comment (nistadev) [reply packages/yoroi-extension/app/components/swap/SwapInput.js:134]: Rn is only for display purposes for the demo, will add intl later on to not block the components development

File: packages/yoroi-extension/app/components/swap/PriceInput.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import { Box, Input, Typography } from '@mui/material';
+
+export default function PriceInput({ label, assets = [], readonly, isLoading }) {
```
Comment (neuodev) [inline packages/yoroi-extension/app/components/swap/PriceInput.js]: Missing types. 
```suggestion

type Props = {|
  label: string,
  assets:  Assets[], 
  readonly?: boolean,
  isLoading?: boolean,
|};

export default function PriceInput({ label, assets = [], readonly, isLoading }) {
```

File: packages/yoroi-extension/app/components/swap/PriceInput.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        color="#000"
+        placeholder="0"
+        bgcolor={readonly ? 'gray.50' : 'common.white'}
+        readOnly={readonly}
+      />
+      <Box sx={{ justifySelf: 'end' }}>
+        <Box height="100%" width="min-content" display="flex" alignItems="center">
+          <Box>{asset1.ticker}</Box>
+          <Box>/</Box>
+          <Box>{asset2.ticker}</Box>
```
Comment (neuodev) [inline packages/yoroi-extension/app/components/swap/PriceInput.js]: I can see you only need the "assets ticker" so you don't need the whole "asset" object. **Component should have access to just what it needs**. 

Also, let's improve the terminology a little bit. For example, for the currency “USDA/ADA”, USDA is the base currency and ADA is the quote currency.

```suggestion
          <Box>{base}</Box>
          <Box>/</Box>
          <Box>{quote}</Box>
```

File: packages/yoroi-extension/app/components/swap/SwapInput.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
+import { useState } from 'react';
+import { Box, Input, Typography } from '@mui/material';
+import { ReactComponent as ChevronIcon } from '../../assets/images/revamp/chevron-icon.inline.svg';
+import { ReactComponent as DefaultTokenImage } from '../../assets/images/revamp/token-default.inline.svg';
+
+export default function SwapInput({
```
Comment (neuodev) [inline packages/yoroi-extension/app/components/swap/SwapInput.js:22]: Please add type definitions.

File: packages/yoroi-extension/app/containers/swap/categories/SwapPage.js [hunk 5]
```diff
@@ -15,144 +13,124 @@ import { PublicDeriver } from '../../../api/ada/lib/storage/models/PublicDeriver
...
+        </Box>
+        <Box sx={{ cursor: 'pointer' }}>
+          <RefreshIcon />
+        </Box>
+      </Box>
 
-  render(): Node {
-    return <>Swap form here</>;
-  }
+      {/* From Field */}
```
Comment (neuodev) [inline packages/yoroi-extension/app/containers/swap/categories/SwapPage.js:48]:
```suggestion
```

File: packages/yoroi-extension/app/containers/swap/categories/SwapPage.js [hunk 6]
```diff
@@ -15,144 +13,124 @@ import { PublicDeriver } from '../../../api/ada/lib/storage/models/PublicDeriver
...
-          updateTheme: { trigger: actions.profile.updateTheme.trigger },
-          updateUnitOfAccount: { trigger: actions.profile.updateUnitOfAccount.trigger },
-        },
-        router: {
-          goToRoute: { trigger: actions.router.goToRoute.trigger },
-        },
-      },
-    });
-  }
+      {/* Clear and switch */}
```
Comment (neuodev) [inline packages/yoroi-extension/app/containers/swap/categories/SwapPage.js:57]:
```suggestion

```

File: packages/yoroi-extension/app/containers/swap/categories/SwapPage.js [hunk 7]
```diff
@@ -15,144 +13,124 @@ import { PublicDeriver } from '../../../api/ada/lib/storage/models/PublicDeriver
...
+          <SwitchIcon />
+        </Box>
+        <Box>
+          <Button variant="tertiary" color="primary">
+            Clear
+          </Button>
+        </Box>
+      </Box>
+
+      {/* To Field */}
```
Comment (neuodev) [inline packages/yoroi-extension/app/containers/swap/categories/SwapPage.js:69]:
```suggestion

```

File: packages/yoroi-extension/app/containers/swap/categories/SwapPage.js [hunk 8]
```diff
@@ -15,144 +13,124 @@ import { PublicDeriver } from '../../../api/ada/lib/storage/models/PublicDeriver
...
+      <SwapInput label="Swap to" asset={{ amount: 0, ticker: '' }} />
+
+      {/* Price between assets */}
```
Comment (neuodev) [inline packages/yoroi-extension/app/containers/swap/categories/SwapPage.js:72]:
```suggestion

```

File: packages/yoroi-extension/app/containers/swap/categories/SwapPage.js [hunk 9]
```diff
@@ -15,144 +13,124 @@ import { PublicDeriver } from '../../../api/ada/lib/storage/models/PublicDeriver
...
+      <Box mt="16px">
+        <PriceInput
+          assets={[
+            { ticker: 'TADA', amount: 20 },
+            { ticker: 'USDA', amount: 5 },
+          ]}
+          readonly={isMarketOrder}
+          label="Market price"
```
Comment (neuodev) [inline packages/yoroi-extension/app/containers/swap/categories/SwapPage.js]: I would suggest using this props structure instead. 

```suggestion
          baseAsset={{ ticker: 'TADA', amount: 20 }}
          quoteAsset={{ ticker: 'USDA', amount: 5 }}
          readonly={isMarketOrder}
          label="Market price"
```

File: packages/yoroi-extension/app/containers/swap/categories/SwapPage.js [hunk 10]
```diff
@@ -15,144 +13,124 @@ import { PublicDeriver } from '../../../api/ada/lib/storage/models/PublicDeriver
...
+        />
+      </Box>
+
+      {/* Slippage settings */}
```
Comment (neuodev) [inline packages/yoroi-extension/app/containers/swap/categories/SwapPage.js:82]:
```suggestion

```

File: packages/yoroi-extension/app/containers/swap/categories/SwapPage.js [hunk 11]
```diff
@@ -15,144 +13,124 @@ import { PublicDeriver } from '../../../api/ada/lib/storage/models/PublicDeriver
...
+        </Box>
+        <Box display="flex" gap="4px" alignItems="center">
+          <Typography variant="body1" color="#000">
+            1%
+          </Typography>
+          <EditIcon />
+        </Box>
+      </Box>
+
+      {/* Available pools */}
```
Comment (neuodev) [inline packages/yoroi-extension/app/containers/swap/categories/SwapPage.js:104]:
```suggestion

```

**Review by neuodev — CHANGES_REQUESTED** [review]

**Comment by nistadev** [comment]
Closing for this: https://github.com/Emurgo/yoroi-frontend/pull/3320

**Pull request closed without merging**